### PR TITLE
Migrate core library and CoreCLR projects to target .NET Standard 1.1

### PR DESCRIPTION
**Pull request by mjcheetham**
As part of the ongoing work to improve ADAL.NET support running on the CoreCLR.

These changes migrate the _ADAL.PCL_ and _ADAL.PCL.CoreCLR_ projects to the new MSBuild 15* format and now target .NET Standard 1.1 (instead of being Portable Class Libraries).
The existing projects (_Desktop_, _Android_, _iOS_, _WinRT_) remain in the MSBuild 14 format.

Due to an issue in Mono (affecting ADAL on Android and iOS) a workaround was previously added to set the `HttpClientHandler.Proxy` property to the default `IWebProxy` (`WebRequest.DefaultWebProxy`).
Because `WebRequest` type does not exist in .NET Standard (until version 2.0) a new platform plugin (`IWebProxyProvider`) was introduced to allow this workaround to continue to work.

\***NB:** MSBuild 15 / VS2017 will be required to build/open the solution.

### CODE REVIEW

File: src/ADAL.PCL.iOS/AdalInitializer.cs [hunk 1]
```diff
@@ -25,24 +25,23 @@
...
-    /// 
-    /// </summary>
     internal static class AdalInitializer
     {
         public static void Initialize()
         {
             PlatformPluginSwitch.DynamicallyLinkAssembly = false;
 
-            PlatformPlugin.InjectDependecies(new WebUIFactory(), new TokenCachePlugin(), new Logger(), new PlatformInformation(), new CryptographyHelper(), new DeviceAuthHelper(), new BrokerHelper());
+            PlatformPlugin.InjectDependecies(
```
Comment (mjcheetham) [inline src/ADAL.PCL.iOS/AdalInitializer.cs]: iOS does not allow dynamic loading of executables. Updated this dependency loading mechanism to pass an instance of the new `IWebProxyProvider` plugin.

File: src/ADAL.PCL/Http/HttpMessageHandlerFactory.cs [hunk 2]
```diff
@@ -42,10 +39,10 @@ internal static HttpMessageHandler GetMessageHandler(bool useDefaultCredentials)
                 return MockHandlerQueue.Dequeue();
             }
 
-            return new HttpClientHandler { UseDefaultCredentials = useDefaultCredentials, Proxy = System.Net.WebRequest.DefaultWebProxy };
+            return new HttpClientHandler { UseDefaultCredentials = useDefaultCredentials, Proxy = PlatformPlugin.WebProxyProvider.GetDefaultWebProxy() };
         }
 
-        private readonly static Queue<HttpMessageHandler> MockHandlerQueue = new Queue<HttpMessageHandler>();
+        private static readonly Queue<HttpMessageHandler> MockHandlerQueue = new Queue<HttpMessageHandler>();
```
Comment (mjcheetham) [inline src/ADAL.PCL/Http/HttpMessageHandlerFactory.cs]: Minor reordering of modifers for consistency :)

**Comment by HSAR** [comment]
Seems AppVeyor doesn't agree with your work.

**Comment by mjcheetham** [comment]
@HSAR, yes.. it won't do since it now requires MSBuild 15 / VS2017 to build.

**Pull request closed without merging**